### DISCUSSION TITLE
Feature request: Smart Charging

### DISCUSSION THREAD

**Discussion by EmanuelLoos** [Divested-Mobile/DivestOS-Build#88] [Ideas] [answered]
iodéOS has a feature called "Smart Charging" (at least on the FP4) which I would like to see in DivestOS. As more and more phones don't have removable batteries anymore I think this could increase the lifetime of phones greatly.
![Screenshot_20220423-112709_Settings](https://user-images.githubusercontent.com/29912739/164891291-79fdb002-79bc-461b-abfb-35d8b1ad7512.png)

**Comment by SkewedZeppelin** [comment, marked as answer]
iodeOS is a proprietary ROM.

There is no standardized way to do that that also has a sane implementation
available.
If you find or write such patches I am happy to merge them.

**Comment by Tvirin** [comment]
Dev Zeelog have add this in latest release, check here https://github.com/zeelog/OTA/releases

It might be helpful, I don't know how he added it I have no knowledge about this I just share

**Reply by SkewedZeppelin** [reply]
I cannot find source code for that OS, making it too proprietary.

**Comment by vlado2222** [comment]
I support this suggestion. On rooted phones can be used [Battery Charge Limit](https://f-droid.org/en/packages/com.slash.batterychargelimit/) app from F-Droid. However rooting is not an option here.

**Comment by thasos** [comment]
Juste found the same issue for [CalyxOS](https://gitlab.com/CalyxOS/calyxos/-/issues/844) and last comment can be interesting :

```
They write: "Useful options from other custom ROMs: Smart charging (disables charging when a given level is reached, to protect battery health).
Maybe they took it from Havoc-OS or CRDroid?
```

It is true : https://github.com/iodeOS/ota#useful-options-from-other-custom-roms

So I searched in both repos but I can't read the matrix :cry: :
- https://github.com/search?q=org%3AHavoc-OS+smart+charging&type=code
- https://github.com/search?q=org%3Acrdroidandroid+smart+charging&type=code

Maybe it can help you ?

**Comment by SkewedZeppelin** [comment]
iodeOS is a proprietary ROM.

There is no standardized way to do that that also has a sane implementation
available.
If you find or write such patches I am happy to merge them.
